src/main/java/Duke.java

```diff
@@ -1,10 +1,63 @@
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+
+    /** List of tasks added by the user */
+    private static final List<Task> tasks = Storage.getData();
+
+    /**
+     * Performs the specified action.
+     *
+     * @param command Command input by the user.
+     * @throws InvalidCommandException If the command cannot be recognised.
+     */
+    public static void runCommand(String command) throws InvalidCommandException {
+        if (command.equals("list")) {
+            TaskList.printList(tasks);
+        } else if (Parser.parseCommand(command).equals("done")) {
+            int index = Parser.parseDoneIndex(command);
+            TaskList.markDone(index, tasks);
+        } else if (Parser.parseCommand(command).equals("todo")) {
+            try {
+                TaskList.addTodo(command, tasks);
+            } catch (InvalidTodoException e) {
+                Ui.printEmptyTodoMessage();
+            }
+        } else if (Parser.parseCommand(command).equals("deadline")) {
+            try {
+                TaskList.addDeadline(command, tasks);
+            } catch (InvalidDateTimeFormatException e) {
+                Ui.printInvalidDateFormatMessage();
+            }
+        } else if (Parser.parseCommand(command).equals("event")) {
+            try {
+                TaskList.addEvent(command, tasks);
+            } catch (InvalidDateTimeFormatException e) {
+                Ui.printInvalidDateFormatMessage();
+            }
+        } else if (Parser.parseCommand(command).equals("delete")) {
+            TaskList.deleteTask(command, tasks);
+        } else if (Parser.parseCommand(command).equals("find")) {
+            TaskList.findTask(command, tasks);
+        } else {
+            throw new InvalidCommandException();
+        }
+    }
```

Comment:
There is quite some content to go through when looking at these if-else statements. A suggestion would be to use switch cases, and perhaps do a single try-catch block that handles different kinds of exceptions.